### PR TITLE
Fix QueryLookupWrapper.getFromSql(Set<FieldKey>)

### PR DESCRIPTION
**Pull request by labkey-matthewb**
#### Rationale
Simplify queries by skipping unused columns.  (previously worked for QuerySelect but not QueryLookupWrapper)

#### Related Pull Requests
* 

#### Changes
*

### CODE REVIEW

File: query/src/org/labkey/query/sql/QueryLookupWrapper.java [hunk 1]
```diff
@@ -267,6 +268,34 @@ public Sort getDefaultSort()
...
+                    lkCol.releaseRef(this);
+                for (var lkCol : _selectedColumns)
+                {
+                    if (fieldKeys.contains(lkCol.getFieldKey()))
+                        lkCol.addRef(this);
+                }
+                // column 0 should still have a reference because it has a reference from _query (see constructor)
+                assert _selectedColumns.get(0).ref.count() > 0;
+            }
+            return super.getFromSQL(alias);
```
Comment (XingY) [inline query/src/org/labkey/query/sql/QueryLookupWrapper.java:296]: if (fieldKeys == null), should it call this.getFromSQL(alias) instead? Kinda strange getFromSQL(alias) behaves differently from getFromSQL(alias, null)

File: query/src/org/labkey/query/sql/QuerySelect.java [hunk 2]
```diff
@@ -1350,16 +1340,28 @@ public SQLFragment getFromSQL(String alias)
...
-                        .collect(Collectors.toSet());
-                    releaseAllSelected(_query);
-                    markAllSelected(new CaseInsensitiveHashSet(names),_query);
+                    markAllSelected(_query);
                 }
                 else
-                    markAllSelected(_query);
+                {
+                    assert !_columns.isEmpty();
+                    if (selectedFieldKeys.isEmpty())
```
Comment (XingY) [inline query/src/org/labkey/query/sql/QuerySelect.java:1350]: I'm having a hard time hitting QuerySelect, can you help me with an example when QuerySelect is used?

File: experiment/src/org/labkey/experiment/api/ExpMaterialTableImpl.java [hunk 3]
```diff
@@ -1081,6 +1081,8 @@ public SQLFragment getFromSQL(String alias, Set<FieldKey> selectedColumns)
     {
         TableInfo provisioned = null == _ss ? null : _ss.getTinfo();
         Set<String> provisionedCols = new CaseInsensitiveHashSet(provisioned != null ? provisioned.getColumnNameSet() : Collections.emptySet());
+        provisionedCols.remove(Column.LSID.name());
```
Comment (XingY) [inline experiment/src/org/labkey/experiment/api/ExpMaterialTableImpl.java:1084]: BTW, I pushed this change to your branch to skip joining to provisioned for LSID and name columns for samples. @labkey-matthewb

File: query/src/org/labkey/query/sql/QuerySelect.java (hunk 2 above)
Comment (labkey-matthewb) [reply query/src/org/labkey/query/sql/QuerySelect.java:1350]: It's possible that we always wrap QuerySelect with QueryLookupWrapper now.  Maybe QueryLookupWrapper.getTableInfo() could check !_hasLookups and then call QuerySelect.getTableInfo().

**Review by XingY — APPROVED** [review]

**Pull request merged**